### PR TITLE
Implement `global.vars`

### LINKED ISSUE #691: `global.vars` is not implemented
See http://www.byond.com/docs/ref/#/DM/vars

Required for TG

### PR DESCRIPTION
**Pull request by ike709**
Closes #691 

Dunno if this is the best way to do it but it works.

### CODE REVIEW

File: DMCompiler/DM/Expressions/LValue.cs [hunk 1]
```diff
@@ -164,4 +164,15 @@ public GlobalField(Location location, DreamPath? path, int id)
         }
     }
 
+    class GlobalVars : LValue
+    {
+        public GlobalVars(Location location, DreamPath? path)
+            : base(location, path) {
```
Comment (wixoaGit) [inline DMCompiler/DM/Expressions/LValue.cs:170]:
```suggestion
        public GlobalVars(Location location)
            : base(location, null) {
```

File: DMCompiler/DM/Visitors/DMVisitorExpression.cs [hunk 2]
```diff
@@ -134,6 +134,10 @@ public void VisitIdentifier(DMASTIdentifier identifier)
             if (globalId != null) {
                 Result = new Expressions.GlobalField(globalIdentifier.Location, DMObjectTree.Globals[globalId.Value].Type, globalId.Value);
                 return;
+            } else if (name == "vars")
+            {
+                Result = new Expressions.GlobalVars(globalIdentifier.Location, null);
```
Comment (wixoaGit) [inline DMCompiler/DM/Visitors/DMVisitorExpression.cs:139]:
```suggestion
                Result = new Expressions.GlobalVars(globalIdentifier.Location);
```

File: OpenDreamRuntime/Objects/DreamList.cs [hunk 3]
```diff
@@ -255,4 +262,58 @@ public override DreamValue GetValue(DreamValue key)
...
+        [Dependency] private readonly IDreamManager _dreamMan = default!;
+
+        private DreamGlobalVars()
+        {
+            IoCManager.InjectDependencies(this);
+        }
+
+        public static DreamGlobalVars Create() {
+            var list = new DreamGlobalVars();
+            list.InitSpawn(new DreamProcArguments(null));
```
Comment (wixoaGit) [inline OpenDreamRuntime/Objects/DreamList.cs:278]: There's no need to initialize a DreamList after creating it.
```suggestion
```

File: OpenDreamRuntime/Procs/DMOpcodeHandlers.cs [hunk 4]
```diff
@@ -368,6 +368,13 @@ static class DMOpcodeHandlers {
             state.Push(new DreamValue(state.ReadString()));
             return null;
         }
+
+        public static ProcStatus? PushGlobalVars(DMProcState state)
+        {
+            var list = state.DreamManager.Globals.GetRange(1, state.DreamManager.Globals.Count - 1);
```
Comment (wixoaGit) [inline OpenDreamRuntime/Procs/DMOpcodeHandlers.cs:374]: This is unused
```suggestion
```

File: OpenDreamRuntime/Objects/DreamList.cs [hunk 5]
```diff
@@ -51,6 +51,13 @@ protected DreamList(int size = 0) : base(null)
             return list;
         }
 
+        public static DreamList Create(List<DreamValue> values)
+        {
+            var list = new DreamList(values.Count);
+            list._values = values;
+            return list;
+        }
+
```
Comment (wixoaGit) [inline OpenDreamRuntime/Objects/DreamList.cs:60]: This is unused
```suggestion
```

**Review by wixoaGit — CHANGES_REQUESTED** [review]

File: OpenDreamRuntime/Objects/DreamList.cs [hunk 6]
```diff
@@ -255,4 +255,57 @@ public override DreamValue GetValue(DreamValue key)
...
+            }
+            var root = _dreamMan.ObjectTree.GetObjectDefinition(DreamPath.Root);
+            if (!root.GlobalVariables.TryGetValue(varName, out var globalId)) {
+                throw new Exception($"Invalid global {varName}");
+            }
+
+            return _dreamMan.Globals[globalId];
+        }
+
+        public override void SetValue(DreamValue key, DreamValue value) {
```
Comment (ike709) [inline OpenDreamRuntime/Objects/DreamList.cs:298]:
```suggestion
        public override void SetValue(DreamValue key, DreamValue value, bool allowGrowth = false) {
```

**Review by wixoaGit — APPROVED** [review]

**Pull request merged**